lib/app/features/chat/providers/conversation_message_actions_provider.c.dart

```diff
@@ -0,0 +1,246 @@
+
+    final id = EventMessage.calculateEventId(
+      tags: tags,
+      createdAt: createdAt,
+      content: encryptedRumor,
+      publicKey: receiverPubkey,
+      kind: BookmarksEntity.kind,
+    );
+
+    final bookmarkMessage = EventMessage(
+      id: id,
+      tags: tags,
+      createdAt: createdAt,
+      content: encryptedRumor,
+      pubkey: receiverPubkey,
+      kind: BookmarksEntity.kind,
+      sig: null,
+    );
+
+    Logger.log('Bookmark message $bookmarkMessage');
+
+    await nostrNotifier.sendEvent(bookmarkMessage, cache: false);
+  }
+
+  Future<void> sendMessageReaction({
+    required String eventId,
+    required String reaction,
+    required String receiverPubkey,
+  }) async {
+    if (eventSigner == null) {
+      throw EventSignerNotFoundException();
+    }
+
+    await _createSealWrapSendReaction(
+      content: reaction,
+      signer: eventSigner!,
+      receiverPubkey: receiverPubkey,
+      kind: PrivateMessageReactionEntity.kind,
+      tags: [
+        ['k', PrivateDirectMessageEntity.kind.toString()],
+        ['p', receiverPubkey],
+        ['e', eventId],
+      ],
+    );
+  }
+
+  Future<void> sendMessageReceivedStatus({
+    required String eventId,
+    required String receiverPubkey,
+  }) async {
+    if (eventSigner == null) {
+      throw EventSignerNotFoundException();
+    }
+
+    await _createSealWrapSendReaction(
+      signer: eventSigner!,
+      content: 'received',
+      receiverPubkey: receiverPubkey,
+      kind: PrivateMessageReactionEntity.kind,
+      tags: [
+        ['k', PrivateDirectMessageEntity.kind.toString()],
+        ['p', receiverPubkey],
+        ['e', eventId],
+      ],
+    );
+  }
+
+  Future<void> sendMessageReadStatus({
+    required String lastMessageId,
+    required String receiverPubkey,
+  }) async {
+    if (eventSigner == null) {
+      throw EventSignerNotFoundException();
+    }
+
+    await _createSealWrapSendReaction(
+      signer: eventSigner!,
+      content: 'read',
+      receiverPubkey: receiverPubkey,
+      kind: PrivateMessageReactionEntity.kind,
+      tags: [
+        ['k', PrivateDirectMessageEntity.kind.toString()],
+        ['p', receiverPubkey],
+        ['e', lastMessageId],
+      ],
+    );
+  }
+
+  Future<NostrEntity?> _createSealWrapSendReaction({
+    required String content,
+    required EventSigner signer,
+    required String receiverPubkey,
+    required List<List<String>> tags,
+    int? kind,
+  }) async {
+    final createdAt = DateTime.now().toUtc();
+
+    final id = EventMessage.calculateEventId(
+      tags: tags,
+      content: content,
+      createdAt: createdAt,
+      publicKey: signer.publicKey,
+      kind: PrivateMessageReactionEntity.kind,
+    );
+
+    final eventMessage = EventMessage(
+      id: id,
+      tags: tags,
+      content: content,
+      createdAt: createdAt,
+      pubkey: signer.publicKey,
+      kind: PrivateMessageReactionEntity.kind,
+      sig: null,
```

Comment:
why the sign is null?

Reply:
Following ION-Connect for EE2E chats we have to send wrapped message to relay and rumor of that message should not be signed, but it is not clear, so I have added signature as for now
![image](https://github.com/user-attachments/assets/13494897-2219-4363-9de6-377d47ef511b)